src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -233,23 +233,212 @@ constexpr std::string_view json_draft_4_metaschema = R"json(
 }
 )json";
 
+/*
+ From https://json-schema.org/draft-06/schema, this is the draft6 metaschema
+ used to validate draft6 json schemas. It's implemented in the draft4 dialect.
+ This is the diff applied to the original metaschema:
```

Comment:
Why is this?

Reply:
Clarified the comment. 

The current rapidjson supports only draft4, so I had to adapt the meta schema.

I added the diff as a reference of what had to change from the official metaschema